### PR TITLE
TRUNK-4855 adding more unit test to PersonTest

### PR DESCRIPTION
**Pull request by ssabada**
Adding more unit test in PersonTest to cover previously uncovered method in Person.java

### CODE REVIEW

**Comment by dkayiwa** [comment]
@ssabada can you add the corresponding should annotations on the methods being tested as at? https://wiki.openmrs.org/display/docs/Unit+Testing+With+at-should+Annotation

**Comment by ssabada** [comment]
I have the should annotations for the methods tested.

File: api/src/main/java/org/openmrs/Person.java [hunk 1]
```diff
@@ -361,6 +361,7 @@ public void setNames(Set<PersonName> names) {
 	 * @see org.openmrs.PersonAttribute
 	 * @should not get voided attributes
 	 * @should not fail with null attributes
+	 * @should
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Person.java]: What is the use of the above should?

File: api/src/test/java/org/openmrs/PersonTest.java [hunk 2]
```diff
@@ -723,6 +723,113 @@ public void getBirthDateTime_shouldReturnNullIfBirthtimeIsNull() throws Exceptio
...
+	 * @see Person#getAttributes(Integer)
+	 * @verifies get attributes based on Integer attributetypeid, null if voided
+	 */
+	@Test
+	public void  getAttributes_shouldGetPersonAttributesBasedOnAttributeTypeIdNullIfVoided() throws Exception {
+		Person person = personHelper(true, 1, 1, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	    Assert.assertEquals(0, person.getAttributes(new Integer(1)).size());
+	}
+
+private Person personHelper(boolean isVoid, int attributeType1, int attributeType2, int attributeType3, String attributeName1, String attributeName2, String attributeName3, String attributeValue1, String attributeValue2, String attributeValue3) {
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/PersonTest.java]: The formatting here looks ugly. Did you run mvn clean install before committing?

Comment (astorga2) [inline api/src/test/java/org/openmrs/PersonTest.java]: Fix this and should annotations.

File: api/src/test/java/org/openmrs/PersonTest.java [hunk 3]
```diff
@@ -723,6 +723,113 @@ public void getBirthDateTime_shouldReturnNullIfBirthtimeIsNull() throws Exceptio
...
+	 	type1.setName(attributeName1);
+	 	type2.setName(attributeName2);
+	 	type3.setName(attributeName3);
+	 	PersonAttribute personAttribute1 = new PersonAttribute(type1, attributeValue1);
+	 	PersonAttribute personAttribute2 = new PersonAttribute(type2, attributeValue2);
+	 	PersonAttribute personAttribute3 = new PersonAttribute(type3, attributeValue3);
+	    
+	    personAttribute1.setVoided(isVoid);
+	    personAttribute2.setVoided(isVoid);
+	    personAttribute3.setVoided(isVoid);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/PersonTest.java]: The formatting here does not look correct. Did you mvn clean install before committing?

**Comment by dkayiwa** [comment]
@ssabada can you look at the formatting in all places and also the empty should annotation?

**Comment by astorga2** [comment]
I have fixed all the formatting and reviewed the should annotations. I also ran mvn clean install FYI.

File: api/src/test/java/org/openmrs/PersonTest.java [hunk 4]
```diff
@@ -723,6 +723,113 @@ public void getBirthDateTime_shouldReturnNullIfBirthtimeIsNull() throws Exceptio
...
+	}
+
+	/**
+	 * @see Person#getAttribute(String)
+	 * @verifies null if all voided, get attribute based on String attributename
+	 */
+	@Test
+	public void  getAttribute_shouldReturnNullIfAttributeNameIsVoided() throws Exception {
+		Person person = personHelper(true, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+		Assert.assertEquals(null, person.getAttribute("name3"));
```
Comment (wluyima) [inline api/src/test/java/org/openmrs/PersonTest.java]: You can user Assert.assertNull instead

File: api/src/test/java/org/openmrs/PersonTest.java [hunk 5]
```diff
@@ -723,6 +723,113 @@ public void getBirthDateTime_shouldReturnNullIfBirthtimeIsNull() throws Exceptio
...
+	}
+
+	/**
+	 * @see Person#getAttribute(Integer)
+	 * @verifies null if all voided, get attribute based on String attributetypeid
+	 */
+	@Test
+	public void  getAttribute_shouldReturnNullWhenExistingPersonAttributeWithMatchingAttributeTypeIdIsVoided() throws Exception {
+		Person person = personHelper(true, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+		Assert.assertEquals(null, person.getAttribute(new Integer(3)));
```
Comment (wluyima) [inline api/src/test/java/org/openmrs/PersonTest.java]: Same here

File: api/src/test/java/org/openmrs/PersonTest.java (hunk 4 above)
Comment (ssabada) [inline api/src/test/java/org/openmrs/PersonTest.java]: I have made the changes for both cases.

Comment (wluyima) [inline api/src/test/java/org/openmrs/PersonTest.java]: You forgot to change this

File: api/src/test/java/org/openmrs/PersonTest.java [hunk 6]
```diff
@@ -723,6 +723,113 @@ public void getBirthDateTime_shouldReturnNullIfBirthtimeIsNull() throws Exceptio
...
+	/**
+	 * @see Person#getAttribute(PersonAttributeType)
+	 * @verifies null if all voided
+	 */
+	@Test
+	public void  getAttribute_shouldReturnNullWhenExistingPersonAttributeTypeIsVoided () throws Exception {
+		Person person = personHelper(true, 1, 2, 3, "name1", "name2", "name3", "value1", "value2", "value3");
+	 	PersonAttributeType type = new PersonAttributeType(new Integer(3));
+	 	type.setName("name3");
+		Assert.assertNull(null, person.getAttribute(type));
```
Comment (wluyima) [inline api/src/test/java/org/openmrs/PersonTest.java]: This is incorrect, it should be

`Assert.assertNull(person.getAttribute(type));`

File: api/src/test/java/org/openmrs/PersonTest.java (hunk 4 above)
Comment (ssabada) [inline api/src/test/java/org/openmrs/PersonTest.java]: Thanks. I completed the changes.

**Pull request merged**